.circleci/workflows.yml

```diff
@@ -0,0 +1,111 @@
+version: 2.1
+parameters:
+  AWSModified:
+    description: 'Used by Path Filtering Orb to determine if changes occurred in the /AWS/ directory'
+    type: boolean
+    default: false
+orbs:
+  aws-cli: circleci/aws-cli@3.1
+  sam: circleci/aws-sam-serverless@3.2.0
+workflows:
+  test_aws_directory_insights:
+    when:
+      and:
+        - and:
+            - not:
+                matches:
+                  pattern: "^master$"
+                  value: << pipeline.git.branch >>
+        - or:
+            - equal: [true, << pipeline.parameters.AWSModified >>]
+    jobs:
+      - build_and_test
+  build_and_deploy_aws_directory_insights:
+    when:
+      and:
+        - equal: [master, << pipeline.git.branch >>]
```

Comment:
should we also require that this task meet the following condition:
`- equal: [true, << pipeline.parameters.AWSModified >>]`
right now I think it would run whenever code is merged into main branch